packages/richtext-lexical/src/exports/react/components/RichText/converter/converters/upload.tsx

```diff
@@ -8,34 +8,29 @@ export const UploadJSXConverter: JSXConverters<SerializedUploadNode> = {
   upload: ({ node }) => {
     // TO-DO (v4): SerializedUploadNode should use UploadData_P4
     const uploadDocument = node as UploadDataImproved
-    if (typeof uploadDocument.value !== 'object') {
+    if (typeof uploadDocument?.value !== 'object') {
       return null
     }
-    const url = uploadDocument.value.url
+
+    const value = uploadDocument.value
+    const url = value.url
 
     /**
      * If the upload is not an image, return a link to the upload
      */
-    if (!uploadDocument.value.mimeType.startsWith('image')) {
+    if (!value.mimeType.startsWith('image')) {
       return (
         <a href={url} rel="noopener noreferrer">
-          {uploadDocument.value.filename}
+          {value.filename}
         </a>
       )
     }
 
     /**
      * If the upload is a simple image with no different sizes, return a simple img tag
      */
-    if (!Object.keys(uploadDocument.value.sizes).length) {
-      return (
-        <img
-          alt={uploadDocument.value.filename}
-          height={uploadDocument.value.height}
-          src={url}
-          width={uploadDocument.value.width}
-        />
-      )
+    if (!Object.keys(value.sizes).length) {
+      return <img alt={value.filename} height={value.height} src={url} width={value.width} />
     }
```

Comment:
Now almost all things derived from value are "any"

Reply:
I have added a type assertion in https://github.com/payloadcms/payload/pull/11292/files.

We could have it fall back to `FileData & TypeWithID`, but I don't know if we _should_. If it has to fall back, that means the collection slug likely doesn't exist, or it isn't an uploads-enabled collection.